gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -496,6 +500,25 @@ private ManagedChannel createSingleChannel() throws IOException {
     return managedChannel;
   }
 
+  /* Remove any provided headers that will also get set by credentials. They will be added as part of the grpc call when performing auth
+   * {@link io.grpc.auth.GoogleAuthLibraryCallCredentials#applyRequestMetadata}.  GRPC does not dedup headers {@link https://github.com/grpc/grpc-java/blob/a140e1bb0cfa662bcdb7823d73320eb8d49046f1/api/src/main/java/io/grpc/Metadata.java#L504} so we must before initiating the call.
+   */
+  private void removeCredentialDuplicateHeaders() {
+    if (headerProvider != null) {
+      headersWithDuplicatesRemoved.putAll(headerProvider.getHeaders());
+    }
+    if (credentials != null) {
+      try {
+        Map<String, List<String>> credentialRequestMetatData = credentials.getRequestMetadata();
+        if (credentialRequestMetatData != null) {
```

Comment:
The other concern I have with this is that this is still going to make a network call to retrieve the metadata. gRPC-Java is going to be doing this as well, so this ends up being 2 network calls, right?

Would it be possible just to remove anything with `API-Keys` or `Authorization` in the header so that we don't need to do the `credentials.getRequestMetadata()`?

Reply:
Thanks Lawrence for bring up this concern! It is definitely a concern if some other types of credentials are making network calls, we don't know that behavior before and I thought it is part of the credentials object already. Do you have links to how other types of credentials would make network calls?

Reply:
> Thanks Lawrence for bring up this concern! It is definitely a concern if some other types of credentials are making network calls, we don't know that behavior before and I thought it is part of the credentials object already. Do you have links to how other types of credentials would make network calls?

I could be wrong on this, but I believe almost every Credential that flows under OauthCredentials is making a network call to retrieve the Access Token. The exception that comes to mind is for SA Credentials using the SSJWT flow.

Oauth2Credential's `asyncFetch`: https://github.com/googleapis/google-auth-library-java/blob/afe4ff9f1f3c406c72ecb66fba50d7463f620467/oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java#L182-L184

Reply:
Lawrence is correct, the headers should be pretty static, @blakeli0 how do you feel about filtering out a static list of authorization headers? If we think this list is going to change in the future, as part of the fast follow I can include  exposing the auth header from the Credentials object.  Or we can remove this entirely until we have that solution?

Reply:
as discussed we will only dedup for apikey credentials for this PR.  action item for @ldetmer to test how other credentials with duplicate headers behave